### PR TITLE
Filter CI on PR contents + Mergify support

### PR DESCRIPTION
**Pull request by abejgonzalez**
**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [x] New feature
- [x] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [x] Other

**Contributor Checklist**:
- [x] Did you set `main` as the base branch?
- [x] Did you delete any extraneous prints/debugging code?
- [x] (If applicable) Did you add documentation for the feature?
- [x] (If applicable) Did you add a test demonstrating the PR?

- [x] (If applicable) Did you mark the PR as "Please Backport"?

**Release Notes**
This PR is copying FireSim's automation for CI which does the following:
- Selectively run certain parts of the CI pipeline based on the files added
- If a backport request is asked, have mergify automate the backport PR (if a label called "Please Backport" is added)
- Use labels to determine what to put in a future changelog (not implemented yet, but figured it would be good to copy this)

Additionally, this PR changes a bit of the wording of the issue/PR templates (in addition to adding new fields specifying what users should do for automation).

Not necessarily part of this PR... but I created a new `1.6.x` branch which should be updated with the "Please backport" label.

### CODE REVIEW

**Comment by abejgonzalez** [comment]
I'll post an issue for someone to optimize the filters a bit more later (have specific file paths trigger specific CI paths).

**Comment by abejgonzalez** [comment]
@Mergifyio rebase

File: .github/workflows/chipyard-run-tests.yml [hunk 1]
```diff
@@ -24,8 +25,42 @@ jobs:
...
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            all:
+              - '**'
+
+            # If any of the files changed match, do a doc build
+            docs: &docs-filter
```
Comment (jerryz123) [inline .github/workflows/chipyard-run-tests.yml:49]: I think this decreases coverage. As this won't catch changes to the read the docs links in the code

**Review by jerryz123 — APPROVED** [review]

**Review by jerryz123 — CHANGES_REQUESTED** [review]
I don't think we can limit when docs are built. Anyways doc building is fast. No need to skip

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  # This rule will need to be updated on minor releases.
+  - name: backport to latest minor release
+    conditions:
+      - merged
+      - base=main
+      - label="Please Backport"
+    actions:
+      backport:
+        branches:
+          - 1.13.x
```
Comment (sagark) [inline .mergify.yml:13]:
```suggestion
          - 1.13.x
```
```suggestion
          - 1.6.x
```

Comment (sagark) [inline .mergify.yml:13]:
```suggestion
          - 1.13.x
```
```suggestion
          - 1.6.x
```

**Review by jerryz123 — APPROVED** [review]

**Pull request merged**